pkg/cli/serve/serve.go

```diff
@@ -131,32 +131,32 @@ func (o *serveOptions) Run() error {
 	log.WithField("URL", o.ocmURL).Debug("OCM URL configured")
 	log.WithField("Service", o.services).Debug("OCM Service configured")
 
-	// create a new router
-	r := mux.NewRouter()
-
-	// Add healthcheck routes
-	healthcheck.Livez().AddRoute(r)
-	healthcheck.Readyz().AddRoute(r)
 	// create new router for metrics
 	rMetrics := mux.NewRouter()
 	rMetrics.Path("/metrics").Handler(promhttp.Handler())
 
-	r.Use(metrics.PrometheusMiddleware)
```

Comment:
Remove this will not call the `PromethuesMiddleware` for the default router. And the metrics cannot be collected as well.

Is this intentional?

Reply:
Thanks a lot for spotting that, that was unintended.

I have pushed a commit to add it back. However, I must ask why the Prometheus middleware would need to be  added to the default router? Shouldn't it be added to the metrics router which is actually serving the metrics endpoint?

I pushed a commit adding it to the metrics router and verified that Prometheus was able to crawl metrics successfully afterwards.

Reply:
The middleware running on default router to proceed and generate the metrics based on the request to default router.
The metrics router just exposes them to metrics port.

Reply:
Fair enough - I have moved it to the default router.